### PR TITLE
openbao: init at 2.0.0

### PR DESCRIPTION
**Pull request by joseph-flinn**
## Description of changes

Add OpenBao as a package with the initial version as 2.0.0.

Homepage: https://openbao.org
GitHub: https://github.com/openbao/openbao
Changelog: https://github.com/openbao/openbao/blob/main/CHANGELOG.md

### Request for specific review
This is my first addition to `nixpkgs` and there are two areas that I would like to call out for specific review.

#### Applicable Tests
Included in this change is a simple inline package test as a stand-in until OpenBao is added as a NixOS service in upcoming PR. A `TODO` has been left with hints to the tests that have yet to be ported over from Vault. Since there is currently only minimal NixOS testing and only the default `checkPhase` testing from `buildGoModule`, I have left the "Tested, as applicable" unchecked for now. Is this sufficient until OpenBao can be added to NixOS as a service?

### Resolves
- https://github.com/NixOS/nixpkgs/issues/300943

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/op/openbao/package.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  tags = [
+    "openbao"
+    "bao"
+  ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/openbao/openbao/version.GitCommit=${src.rev}"
```
Comment (cipherboy) [inline pkgs/by-name/op/openbao/package.nix:45]: As a note, I think you're short two more flags:

- `-X github.com/openbao/openbao/version.fullVersion=...`
- `-X github.com/openbao/openbao/version.BuildDate=...`

You could probably compute build date from the commit date if you want ([we use the build's actual date upstream](https://github.com/openbao/openbao/blob/main/.goreleaser-template.yaml#L29)).

hth!

File: pkgs/by-name/op/openbao/package.nix [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            gawk
+            glibc
+          ]
+        }
+    '';
+
+  # TODO: Enable the NixOS tests after adding OpenBao as a NixOS service in an upcoming PR and
+  # adding NixOS tests
+  #
+  # passthru.tests = { inherit (nixosTests) vault vault-postgresql vault-dev vault-agent; };
```
Comment (cipherboy) [inline pkgs/by-name/op/openbao/package.nix:67]: One other note, it looks like `vault-postgres` tests with Postgres as a storage backend, which this release does not support. OpenBao 2.1.0 [will support Postgres](https://github.com/openbao/openbao/pull/467), barring unforseen issues.

**Comment by liberodark** [comment]
Hi,

Any news on your pr ?

Best Regards

File: pkgs/by-name/op/openbao/package.nix (hunk 1 above)
Comment (cipherboy) [reply pkgs/by-name/op/openbao/package.nix:45]: We recently did some testing of this at GitLab and found BuildDate was completely ignored so I think you're fine leaving it off. But I think fullVersion is definitely necessary.

**Comment by brianmay** [comment]
I created a pull request with some updates. #354366

Will close that PR if/when this is updated.

**Comment by joseph-flinn** [comment]
@brianmay I no longer have the time to be the core maintainer of the openbao nixpkg. I'll close this PR in favor for yours: https://github.com/NixOS/nixpkgs/pull/354366

**Pull request closed without merging**